mod/github_test.go

```diff
@@ -42,25 +63,66 @@ func TestGitHubMgrGet(t *testing.T) {
 }
 
 func TestGitHubMgrFind(t *testing.T) {
-	githubmod := &githubMgr{}
+	cacheDir := ".pkgcache"
+	repo := "github.com/foo/bar"
+	filea := "filea"
+	fileb := "fileb"
+	tagRef := "v0.2.0"
+	masterRef := "v0.0.0-41f04d3bba15"
+	tagRepoDir := "github.com/foo/bar@v0.2.0"
+	masterRepoDir := "github.com/foo/bar@v0.0.0-41f04d3bba15"
+
+	githubmod := &githubMgr{cacheDir: cacheDir}
 	testMods := Modules{}
-	local := &Module{Name: "local", Version: "v0.0.1"}
-	mod1 := &Module{Name: "remote", Version: "v1.0.0-41f04d3bba15"}
-	mod2 := &Module{Name: "remote", Version: "v0.2.0"}
-	testMods.Add(local)
-	testMods.Add(mod1)
-	testMods.Add(mod2)
-
-	assert.Equal(t, local, githubmod.Find("local/filename", "v0.0.1", &testMods))
-	assert.Equal(t, local, githubmod.Find("local/filename2", "v0.0.1", &testMods))
-	assert.Equal(t, local, githubmod.Find(".//local/filename", "v0.0.1", &testMods))
-	assert.Equal(t, local, githubmod.Find("local", "v0.0.1", &testMods))
-	assert.Nil(t, githubmod.Find("local2/filename", "v0.0.1", &testMods))
-
-	assert.Nil(t, githubmod.Find("local/filename", MasterBranch, &testMods))
-
-	assert.Equal(t, mod2, githubmod.Find("remote/filename", "v0.2.0", &testMods))
-	assert.Nil(t, githubmod.Find("remote/filename", "v1.0.0", &testMods))
+	tagMod := &Module{
+		Name:    repo,
+		Version: tagRef,
+		Dir:     tagRepoDir,
+	}
+	masterMod := &Module{
+		Name:    repo,
+		Version: masterRef,
+		Dir:     masterRepoDir,
+	}
+	testMods.Add(tagMod)
+	testMods.Add(masterMod)
+
+	monkey.Patch(FileExists, func(filename string, _ bool) bool {
```

Comment:
Nit: `monkey` seems like a really cool package! But I can't help but feel that this mocking might be better accomplished using an afero.NewMemMapFS() that we add some files to and is passed into SyslMod. Might need to modify a lot more though, so the current implementation is also fine, just mentioning for future reference

Reply:
Tried this before, the point here is that `FileExists` finds files directly by using `os` package, so `afero.NewMemMapFS()` doesn't work.